spec/jobs/repo_setup_job_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe RepoSetupJob, type: :job do
+  let(:organization) { classroom_org }
+  let(:user)         { classroom_student }
+
+  let(:assignment)      { create(:assignment, organization: organization) }
+  let(:assignment_repo) { create(:assignment_repo, assignment: assignment, user: user, github_repo_id: 848) }
+
+  let(:unconfigured_repo) { stub_repository("template") }
+  let(:configured_repo)   { stub_repository("configured-repo") }
+
+  let(:repo_access)  { RepoAccess.create(user: user, organization: organization) }
+
+  let(:grouping)     { create(:grouping, organization: organization) }
+  let(:group)        { Group.create(title: "Group 1", grouping: grouping) }
+
+  let(:group_assignment) do
+    create(:group_assignment,
+           grouping: grouping,
+           title: "Learn JavaScript",
+           organization: organization,
+           public_repo: true,
+           starter_code_repo_id: 1_062_897)
+  end
+
+  let(:group_assignment_repo) do
+    group.repo_accesses << repo_access
+    GroupAssignmentRepo.create(group_assignment: group_assignment, group: group)
+  end
+
+  before do
+    Octokit.reset!
+  end
+
+  after(:each) do
+    group.destroy
```

Comment:
Why are we destroying everything after each spec? Rspec will recover the state of the application before the `it` block ran for each spec, they don't influence eachother.